### ISSUE TITLE
css Props helper function syntax similar to html helper functions

### ISSUE THREAD

**Issue by B-Teague** [cristianbote/goober#382]
I was writing my own CSS-in-JS library [morpho](https://github.com/B-Teague/morpho) and I came across goober that has a lot of the same features I wanted including using JSON objects for cssProps.  I like goober's implementation of vendor prefixing better, and it already had SSR implemented.  Instead of trying to "reinvent the wheel", I would like to contribute to goober instead.

I'd like to add a new feature for goober that I was planning for morpho that is similar to [hyperapp](https://github.com/JorgeBucaran/hyperapp)'s feature [html](https://github.com/jorgebucaran/hyperapp/tree/main/packages/html) in order to take advantage of VSCode intellisense

Something like this syntax:

```javascript
import { combo, display, flex, border, px } from 'goober/cssProps'
import { FLEX, RED, GREEN, SOLID } from 'goober/cssProps/constants'

styled('span')(
  combo('&', [
    display(FLEX),
    border(px(3), SOLID),
    flex(1),
    color(RED),
    combo('&:hover', [
      color(GREEN)
   ])
);
```

Combo (can use a different name) is short for combination, and has to be called at the root level and for each combinator, pseudo selector, or media query.

Could also create constants for common used pseudo selectors:
```javascript
combo(ON_HOVER, [
  color(GREEN)
])
```

I would write all the code for this feature.
Do you have interest in using / supporting this coding style?
Feedback and suggestions are welcome!

**Comment by cristianbote** [comment]
Hi @B-Teague! 👋 Thank you so much for opening this thread! Kudos on morpho looks great!

TL;DR; totally interested!

Neat! Haven't seen this kind of style composition before. The interesting thing is I believe as long as `combo` returns an object or an array this should work out of the box since the `css` function supports objects, tagged templates, functions or arrays.

```js
const Span = styled('span')(() => [
  { display: 'flex' },
  { color: 'dodgerblue' }
]));
```

So as long as `combo()` returns what `css` needs I don't see any issues or problems. I believe this would be something useful for `goober`'s ecosystem and since it's gonna be a separate entrypoint `goober/cssProps` or any other name, I think it would affect the bundle size only if it's being used.

**Comment by B-Teague** [comment]
Awesome!  I'll send a PR your way once I'm done testing.

**Comment by cristianbote** [comment]
I believe the discussion will happen onto the PR, right @B-Teague? Can we close this issue?
